docs/dev/new_check_howto.md

```diff
@@ -88,187 +91,166 @@ A Check is a Python class with the following requirements:
+Checks are organized in regular Python packages under the `datadog_checks` namespace, so your code should live under `awesome/datadog_checks/awesome`. The only requirement is that the name of the package has to be the same as the check name. There are no particular restrictions on the name of the Python modules within that package, nor on the name of the class implementing the check.
 
 ### Implement check logic
 
-Let's say we want to create a Service Check named `my_check` that checks for a string on a web page. It will result in `OK` if the string is present, `WARNING` if the page is accessible but the string was not found, and `CRITICAL` if the page is inaccessible.
+Let's say we want to create a Service Check named `awesome.search` that searches for a string on a web page. It will result in `OK` if the string is present, `WARNING` if the page is accessible but the string was not found, and `CRITICAL` if the page is inaccessible.
 
 The code would look something like this:
 
 ```python
 import requests
 
-from datadog_checks.checks import AgentCheck
-from datadog_checks.errors import CheckException
+from datadog_checks.base import AgentCheck, ConfigurationError
 
 
-# MyCheck derives from AgentCheck, and provides the required check method.
-class MyCheck(AgentCheck):
+class AwesomeCheck(AgentCheck):
+    """AwesomeCheck derives from AgentCheck, and provides the required check method."""
+
     def check(self, instance):
         url = instance.get('url')
         search_string = instance.get('search_string')
 
-        # It's a good idea to do some basic sanity checking. Try to be as
-        # specific as possible, with the exceptions; you can fall back to
-        # CheckException when in doubt though.
+        # It's a very good idea to do some basic sanity checking.
+        # Try to be as specific as possible with the exceptions.
         if not url or not search_string:
-            raise CheckException("Configuration error, please fix my_check.yaml")
+            raise ConfigurationError('Configuration error, please fix awesome.yaml')
 
         try:
-            r = requests.get(url)
-            r.raise_for_status()
-            if search_string in r.text:
-                # Page is accessible and the string is present.
-                self.service_check('my_check.all_good', self.OK)
-            else:
-                # Page is accessible but the string was not found.
-                self.service_check('my_check.all_good', self.WARNING)
+            response = requests.get(url)
+            response.raise_for_status()
+
+        # Something went horribly wrong
         except Exception as e:
-            # Something went horribly wrong. Ideally we'd be more specific...
-            self.service_check('my_check.all_good', self.CRITICAL, e)
+            # Ideally we'd use a more specific message...
+            self.service_check('awesome.search', self.CRITICAL, message=str(e))
+
+        # Page is accessible
+        else:
+
```

Comment:
whitespace?

Reply:
I thought it would be better for showing/demoing. I can remove if you want

Reply:
People _will_ use this as a basis for their own code, so we might as well enforce proper style right from the get-go, imho.